Haim/issues fix

haimkastner opened the pull request:
1) Cache mechanism #20 .
2) Logger.

haimkastner committed:
Implementation of a simple caching mechanism.
see #20

haimkastner committed:
Use logger.
- also remove log when auth fail.

haimkastner committed:
minor schemas changes.

haimkastner committed:
Remove 'moment' dependency

eranshmil commented on src/core/cache.ts:
```diff
@@ -0,0 +1,34 @@
+    return this._nodeCache.get(key);
+  }
+
+  /** Save or set value map by key to cache. */
```
JSDoc? :)

haimkastner replied:
Yes... I fixed it in 8ae4d3da051b02ae74eaf205572a24405a8b6cef

eranshmil commented on src/security/authentication.ts:
```diff
@@ -1,13 +1,10 @@
-  stdTTL: 60 * 60 * 2, // Every 2 hours reread key from DB.
-  checkperiod: 60 * 30 // Clear old cache every 30 minutes.
-});
+const reportersAuthCache = new Cache(2 * 60 * 60);
```
Cache for this one isn't necessary, the report route wouldn't be called so frequently.

haimkastner replied:
I removed it in 8ae4d3da051b02ae74eaf205572a24405a8b6cef, but it was fun.

eranshmil commented on src/models/user-status.model.ts:
```diff
@@ -6,30 +6,29 @@ import { Platform, Status, Reason } from '../core';
+  @PrimaryGeneratedColumn() private id: number;
 
-  @Column({ name: 'user_id', type: 'varchar', length: 30 })
+  @Column({ name: 'user_id', type: 'varchar', length: 30, nullable: false })
```
You can add the `postId` also in this PR.

haimkastner replied:
Implemented with 581bf277cfcd6fa39ab547044d2b00ccc8a109f1

eranshmil commented on src/controllers/user-statuses-controller.ts:
```diff
@@ -21,47 +21,29 @@ export class UserStatusesController extends Controller {
+    /** Try to retrieve users status from the cache before reading data from the DB. */
+    const usersStatusMap: UserStatusMap = {};
+    for (const userId of userIds) {
+      const cacheResult = await usersCache.get(`${platform}:${userId}`);
```
For the next version, we should change that logic to use `mget` and some other structure for the values.
What do you think?

eranshmil committed:
chore: fix heroku deploy

eranshmil committed:
chore: redirect to website from root path

haimkastner committed:
Extend schemas. see #22
- Add `post_id` column.
- Add `unique` key to `user_statuses` table.
- Add submit a report as `DUPLICATE` supports.

haimkastner committed:
Remove unnecessary cache.
Add JSdoc to cache.

eranshmil committed:
fix: typo and minor ocd changes

haimkastner committed:
Fix check report exsit query.
And changes in user model.

eranshmil committed:
chore: package scripts cleanup and more minor changes
